modules/dag-shared/src/main/scala/org/tessellation/dag/block/BlockValidator.scala

```diff
@@ -0,0 +1,153 @@
+package org.tessellation.dag.block
+
+import cats.Applicative
+import cats.data.Validated.{Invalid, Valid}
+import cats.data.{NonEmptyList, ValidatedNel}
+import cats.effect.Async
+import cats.syntax.applicative._
+import cats.syntax.flatMap._
+import cats.syntax.foldable._
+import cats.syntax.functor._
+import cats.syntax.show._
+import cats.syntax.traverse._
+import cats.syntax.validated._
+
+import scala.util.control.NoStackTrace
+
+import org.tessellation.dag.block.BlockValidator._
+import org.tessellation.dag.transaction.TransactionValidator
+import org.tessellation.dag.transaction.TransactionValidator.TransactionValidationError
+import org.tessellation.dag.transaction.filter.Consecutive
+import org.tessellation.dag.types.{BlockReference, DAGBlock}
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.transaction.{Transaction, TransactionReference}
+import org.tessellation.security.SecurityProvider
+import org.tessellation.security.signature.Signed
+
+abstract class BlockValidator[F[_]](transactionValidator: TransactionValidator[F], consensusPeersCount: Int) {
```

Comment:
maybe `facilitators` instead of `consensus peers`?

Reply:
We also have the notion of facilitators (participating) and owner. Then it could be misleading. But I can actually make it more precise and go with `requiredUniqueBlockSigners` or something similar.

Reply:
but I think it's strictly related to the number of facilitators during the consensus round 🤔 I mean, that should be the same value. anyway, `requiredUniqueBlockSigners` would be fine :+1: name is descriptive enough and one can pass the same configuration value here and to the consensus :+1: